docs/konnectors_workflow_example.md

```diff
@@ -0,0 +1,368 @@
+      }
+    }
+  }
+}
+```       
+
+9 - SettingsApp add a Reference from konnector to folder to prevent folder destruction
+
+```http
+POST /files/123-selected-folder-id-123/relationships/referenced_by
+```
+```json
+{
+  "data": [
+    {
+      "type": "io.cozy.konnectors",
+      "id": "io.cozy.konnectors/trainlines"
+    }
+  ]
+}
+```
+
+10 - SettingsApp runs the konnector the first time
+
+```http
+POST /jobs/queue/konnector
+```
+```json
+{
+  "data": {
+    "attributes": {
+       "options": {
+       "priority": 10,
+       "timeout": 180,
+       "max_exec_count": 1
+      }
+    },
+    "arguments": {
+      "konnector": "io.cozy.konnectors/trainlines",
+      "account": "123-account-id-123",
+      "folderToSave": "123-selected-folder-id-123"
+    }
+  }
+}
+```
+```http
+HTTP/1.1 200 OK
+```
+```json
+{
+  "data": {
+    "id":"789-job-id-789",
+    "type":"io.cozy.jobs",
+    "attributes": {
+      "worker": "konnector",
+      "options": {
+        "priority": 3,
+        "timeout": 60,
+        "max_exec_count": 3
+      },
+      "arguments": {
+        "konnector": "io.cozy.konnectors/trainlines",
+        "account": "123-account-id-123",
+        "folderToSave": "123-selected-folder-id-123"
+      },
+      "state": "running",
+      "try_count": 1,
+      "queued_at": "2016-09-19T12:35:08Z",
+      "started_at": "2016-09-19T12:35:08Z",
+      "errors": [],
+      "output": {}
+    }
+  }
+}
+```
+
+11 - SettingsApp follow konnector status to check if all is properly configured.
+It uses the [realtime](./realtime.md) to subscribes to changes on `789-job-id-789` and is therefore informed of it's status / errors / ect ...
+
+
+**TODO** Look at current konnectors sources to defines a protocol between konnectors and SettingsApp to display the nice progress modal.
+  - [ ] 250 events imported
+  - [ ] 150 / 3500 contacts importing
+  - [ ] ...
+
+**TODO** there should be some persistence for jobs error / status
+
+
+12 - SettingsApp creates a trigger to setup the konnector recurence
+
+```http
+POST /jobs/io.cozy.triggers
+```
+```json
+{
+  "data": {
+    "attributes": {
+      "type": "@cron",
+      "arguments": "0 0 0 0 1 1 ",
+      "worker": "konnector",
+      "worker_arguments": {
+        "konnector": "trainlines",
+        "account": "5165621628784562148955",
+        "folderToSave": "877854878455",
+      }
+    }
+  }
+}
+```
+
+If the user wants to use several account, Settings can setup several triggers for the same konnector & various accounts.
+
+
+### Relations and DELETE CASCADE
+
+- When a konnector is deleted, its triggers should be deleted
+- When an account is deleted, its triggers should be deleted
+
+
+**TODO** Should stack validate konnector value on trigger creation / to be extended to all workers pre-validating worker arguments ?
+
+
+## Konnector Worker specs
+
+- Start the konnector through Rkt, passing token as ENV variable
+- Pass the rest of worker_arguments to the konnector (ENV / stdin ?)
```

Comment:
I proposed the use COZY_FIELDS ENV variable as a JSON string. You can take a look at : https://github.com/cozy/cozy-stack/blob/master/docs/konnectors_design.md#details-about-running-a-konnector